SpotifySearch/SpotifySearchTableViewController.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  SpotifySearchTableViewController.swift
+//  SpotifySearch
+//
+//  Created by John Gabriel Breshears on 10/28/16.
+//  Copyright © 2016 John Gabriel Breshears. All rights reserved.
+//
+
+import UIKit
+
+class SpotifySearchTableViewController: UITableViewController {
+
+    
+    var allAlbums = [Album]()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        APIRequestManager.manager.getAlbumData { (data: Data?) in
+            if let unwrappedAlbumData = Album.album(from: data!) {
+                self.allAlbums = unwrappedAlbumData
+                DispatchQueue.main.async {
+                    self.tableView.reloadData()
+                }
+            }
+        }
+    }
+    
+
+
+    // MARK: - Table view data source
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        // #warning Incomplete implementation, return the number of sections
+        return 1
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        // #warning Incomplete implementation, return the number of rows
+        return allAlbums.count
+    }
+
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: "AlbumCell", for: indexPath) 
+
+        let album = allAlbums[indexPath.row]
+        let name = album.albumName
+        
+        cell.textLabel?.text = name
+        
+        cell.imageView?.downloadImage(urlString: album.imageUrl)
+        
+        APIRequestManager.manager.getAlbumData { (data: Data?) in
```

Comment:
Why do have this code? This is messing up your tableview cells and causing them to repeat somehow. 

Your extension on imageView already makes an asynchronous call to fetch the album image and reloads tableview on main thread. 

This code makes a complete fetch of your ENTIRE data at each tableview cell and then reloads tableview again. Can you see how inefficient that would make your code run?

Good job though, Gabriel!